### PR TITLE
ocamlPackages.pfff: init at 2021-12-14

### PR DESCRIPTION
**Pull request by jtojnar**
> [pfff](https://github.com/returntocorp/pfff/) is a set of tools and APIs to perform static analysis, code visualizations, code navigations, or style-preserving source-to-source transformations such as refactorings on source code. There is good support for Javascript, Python, C, Java, Go, and PHP. There is also preliminary support for other languages such as C++, Ruby, Rust, C#, Html, CSS, Erlang, Lisp, Haskell, Skip, and SQL. There is also very good support for OCaml code so that the framework can be used on the code of pfff itself.

Unfortunately it looks like it has been quite gutted.

### CODE REVIEW

File: pkgs/development/ocaml-modules/easy_logging_yojson/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  useDune2 = true;
+  strictDeps = true;
+
+  meta = with lib; {
+    description = "Configuration loader for easy_logging with yojson backend";
+    homepage = "https://sapristi.github.io/easy_logging/";
+    license = licenses.mpl20;
+    maintainers = [ ];
+    platforms = ocaml.meta.platforms or [];
```
Comment (ambroisie) [inline pkgs/development/ocaml-modules/easy_logging_yojson/default.nix:32]: Already set by `buildDunePackage`.

File: pkgs/development/ocaml-modules/pfff/default.nix [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    dune install --prefix $out --libdir $OCAMLFIND_DESTDIR
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Tools and APIs for program analysis, code visualization, refactoring";
+    homepage = "https://github.com/returntocorp/pfff/wiki";
+    license = licenses.lgpl21Only;
+    maintainers = [ ];
+    platforms = ocaml.meta.platforms or [];
```
Comment (ambroisie) [inline pkgs/development/ocaml-modules/pfff/default.nix:88]: Already set by `buildDunePackage`.

File: pkgs/development/ocaml-modules/easy_logging_yojson/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  };
```
Comment (ambroisie) [inline pkgs/development/ocaml-modules/easy_logging_yojson/default.nix:33]:
```suggestion
  meta = easy_loggin.meta // {
    description = "Configuration loader for easy_logging with yojson backend";
  };
```

File: pkgs/development/ocaml-modules/easy_logging/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  useDune2 = true;
+  strictDeps = true;
+
+  meta = with lib; {
+    description = "Module to log messages";
+    homepage = "https://sapristi.github.io/easy_logging/";
+    license = licenses.mpl20;
+    maintainers = [ ];
+    platforms = ocaml.meta.platforms or [];
```
Comment (ambroisie) [inline pkgs/development/ocaml-modules/easy_logging/default.nix:35]: Already set by `buildDunePackage`.

File: pkgs/development/ocaml-modules/easy_logging/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  nativeBuildInputs = [
+    ocaml
+  ];
+
+  propagatedBuildInputs = [
+    calendar
+  ];
+
+  useDune2 = true;
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/easy_logging/default.nix:27]:
```suggestion
  duneVersion = "2";
```

File: pkgs/development/ocaml-modules/easy_logging/default.nix [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    owner = "sapristi";
+    repo = "easy_logging";
+    rev = "v${version}";
+    sha256 = "Xy6Rfef7r2K8DTok7AYa/9m3ZEV07LlUeMQSRayLBco=";
+  };
+
+  nativeBuildInputs = [
+    ocaml
+  ];
+
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/easy_logging/default.nix:22]:
```suggestion
```

added by `buildDunePackage`

File: pkgs/development/ocaml-modules/easy_logging_yojson/default.nix [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+}:
+
+buildDunePackage rec {
+  pname = "easy_logging_yojson";
+  inherit (easy_logging) version src;
+
+  nativeBuildInputs = [
+    ocaml
+  ];
+
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/easy_logging_yojson/default.nix:17]:
```suggestion
```

added by `buildDunePackage`

File: pkgs/development/ocaml-modules/easy_logging_yojson/default.nix [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  propagatedBuildInputs = [
+    ppx_deriving
+    ppx_deriving_yojson
+    easy_logging
+  ];
+
+  useDune2 = true;
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/easy_logging_yojson/default.nix:24]:
```suggestion
  duneVersion = "2";
```

File: pkgs/development/ocaml-modules/pfff/default.nix [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+  nativeBuildInputs = [
+    ocaml
+    opam
+    perl
+    dune_2
+    menhir
+    grain_dypgen
+  ];
+
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/pfff/default.nix:42]:
```suggestion
  nativeBuildInputs = [
    opam
    perl
    menhir
    grain_dypgen
  ];
```

Already added by `buildDunePackage`

File: pkgs/development/ocaml-modules/pfff/default.nix [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+    ppxlib
+    uutf
+    uucp
+    alcotest
+    ansiterminal
+    easy_logging
+    easy_logging_yojson
+  ];
+
+  useDune2 = true;
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/pfff/default.nix:58]:
```suggestion
  duneVersion = "2";
```

File: pkgs/development/ocaml-modules/pfff/default.nix [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    runHook preCheck
+    dune runtest ''${enableParallelBuilding:+-j $NIX_BUILD_CORES}
+    runHook postCheck
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    dune install --prefix $out --libdir $OCAMLFIND_DESTDIR
+    runHook postInstall
+  '';
```
Comment (Et7f3) [inline pkgs/development/ocaml-modules/pfff/default.nix:81]: seems like what `buildDunePackage` do why redefine it ?

**Review by Et7f3 — COMMENTED** [review]
I don't know why this PR is in draft. If it is because you are hesitant to adding pfff. You can still open smaller PR for `easy_logging` and `easy_logging_yojson` that are present in opam repository.

**Comment by jtojnar** [comment]
Sorry, I only packaged it to try it out and it did not do what I wanted. Feel free to take over if you need it.

**Pull request closed without merging**